allennlp/models/simple_tagger.py

```diff
@@ -0,0 +1,137 @@
+from typing import Dict
+
+import numpy
+import torch
+from torch.nn.modules.linear import Linear
+from torch.nn.modules import LSTM
+import torch.nn.functional as F
+
+from ..training.model import Model
+from ..modules.embeddings import Embedding
+from ..modules.time_distributed import TimeDistributed
+from ..data.vocabulary import Vocabulary
+from ..data.fields.text_field import TextField
+
+
+class SimpleTagger(Model):
+    """
+    This ``SimpleTagger`` simply encodes a sequence of text with some number of stacked
+    ``seq2seq_encoders``, then predicts a tag for each token in the sequence.
+
+    Parameters
+    ----------
+    vocabulary : Vocabulary, required
+        A Vocabulary, required in order to compute sizes for input/output projections.
+    embedding_dim : int, optional (default = 100)
+        The dimensionality of the embedding space used to embed the input sequence.
+    hidden_size : int, optional (default = 200)
+        The dimensionality of the hidden state of the LSTM encoder.
+    num_layers : int, optional (default = 2)
+        The number of stacked LSTM encoders to use.
+    """
+
+    def __init__(self,
+                 vocabulary: Vocabulary,
+                 embedding_dim: int = 100,
+                 hidden_size: int = 200,
+                 num_layers: int = 2):
+        super(SimpleTagger, self).__init__()
+
+        self.vocabulary = vocabulary
+        self.embedding_dim = embedding_dim
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_classes = self.vocabulary.get_vocab_size("tags")
+
+        self.embedding = Embedding(self.vocabulary.get_vocab_size("tokens"),
+                                   self.embedding_dim)
+        # TODO(Mark): support masking once utility functions are merged.
+        self.stacked_encoders = LSTM(self.embedding_dim,
+                                     self.hidden_size,
+                                     self.num_layers,
+                                     batch_first=True)
+        self.tag_projection_layer = TimeDistributed(Linear(self.hidden_size,
+                                                           self.num_classes))
+        self.sequence_loss = torch.nn.CrossEntropyLoss()
+
+    def forward(self,  # pylint: disable=arguments-differ
+                sequence_tokens: torch.IntTensor,
+                sequence_tags: torch.IntTensor = None):
+        """
+        Parameters
+        ----------
+        sequence_tokens : torch.LongTensor, required
```

Comment:
Thanks. There are annoying differences in pytorch with functions which should take both, but don't. For instance, you can only index other tensors with `LongTensors`, not `IntTensors`.